### PR TITLE
autogen

### PR DESCRIPTION
**Pull request by jburel**
read config from cli

Related https://github.com/openmicroscopy/openmicroscopy/pull/6020

The autogen Job has also been modified to export the pythonpath

### CODE REVIEW

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-docs-autogen/1099/ is green

File: omero/autogen_db_version.py [hunk 1]
```diff
@@ -16,12 +17,11 @@ def get_mmp(sqlfile):
 serverdir = sys.argv[1]
```
Comment (mtbc) [inline omero/autogen_db_version.py]: @sbesson points out that using `path(sys.argv[1])` should make https://github.com/openmicroscopy/openmicroscopy/pull/6020 unnecessary.

Comment (jburel) [reply omero/autogen_db_version.py]: it does not. I have tried ``os.path.abspath`` for example and it still fails

Comment (joshmoore) [reply omero/autogen_db_version.py]: You need `import path` from https://github.com/openmicroscopy/openmicroscopy/blob/d66d25f33582d018551209fc33a8a076c3f6ebb8/components/tools/OmeroPy/src/omero/cli.py#L38

Comment (jburel) [reply omero/autogen_db_version.py]: That path!!

**Comment by jburel** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-docs/1434/ is green without https://github.com/openmicroscopy/openmicroscopy/pull/6020 in the build

File: omero/autogen_db_version.py [hunk 2]
```diff
@@ -5,23 +5,25 @@
...
 
 def get_mmp(sqlfile):
     m = re.search('.*/?OMERO(\d+)\.(\d+)(\w*)__(\d+)', sqlfile)
     mmp = (int(m.group(1)), int(m.group(2)), m.group(3), int(m.group(4)))
     return mmp
 
 
-serverdir = sys.argv[1]
+serverdir = path(sys.argv[1])
+print serverdir
```
Comment (mtbc) [inline omero/autogen_db_version.py]: Want to retain the `print`?

Comment (jburel) [reply omero/autogen_db_version.py]: oops

Comment (jburel) [reply omero/autogen_db_version.py]: sorted

**Comment by jburel** [comment]
@mtbc ok to merge now?
That will help for the rest of the doc

**Comment by mtbc** [comment]
Current failure in OMERO-DEV-merge-docs-autogen looks to me due to #1948, not this PR.

File: omero/autogen_db_version.py [hunk 3]
```diff
@@ -5,23 +5,25 @@
...
     m = re.search('.*/?OMERO(\d+)\.(\d+)(\w*)__(\d+)', sqlfile)
     mmp = (int(m.group(1)), int(m.group(2)), m.group(3), int(m.group(4)))
     return mmp
 
 
-serverdir = sys.argv[1]
+serverdir = path(sys.argv[1])
+
 required = {'omero.db.version': None, 'omero.db.patch': None,
             'versions.bioformats': None}
```
Comment (mtbc) [inline omero/autogen_db_version.py]: Where does `versions.bioformats` come from now?

File: omero/autogen_db_version.py [hunk 4]
```diff
@@ -5,23 +5,25 @@
...
-    for line in f.readlines():
-        m = re.match('([\w\.]+)=(.*)', line.strip())
-        if m and m.group(1) in required:
-            required[m.group(1)] = m.group(2)
-            print '%s = "%s"' % (m.group(1).replace('.', '_'), m.group(2))
+cli = CLI()
+property_lines = cli.get_config_property_lines(serverdir)
+for property in PropertyParser().parse_lines(property_lines):
+    if property.key in required:
+        required[property.key] = property.val
```
Comment (mtbc) [inline omero/autogen_db_version.py:25]: (Pity you can't just use `PropertyParser`'s `data()` getter but it chokes on `product.name=OMERO.server`.)

**Review by mtbc — APPROVED** [review]
Some small comments but no blockers.

File: omero/autogen_db_version.py (hunk 4 above)
Comment (joshmoore) [reply omero/autogen_db_version.py:25]: Do we know _why_ it chokes?

Comment (mtbc) [reply omero/autogen_db_version.py:25]: https://github.com/openmicroscopy/openmicroscopy/blob/v5.5.0-m5/components/tools/OmeroPy/src/omero/install/config_parser.py#L320

Comment (joshmoore) [reply omero/autogen_db_version.py:25]: Gotcha. Think that property can now go away ;)

**Review by mtbc — APPROVED** [review]

File: omero/autogen_db_version.py (hunk 4 above)
Comment (jburel) [reply omero/autogen_db_version.py:25]: the property was needed for one of the jobs. this will need to be reviewed before we remove it.

**Pull request merged**